### DISCUSSION TITLE
Can you specify raw SQL for entity property onUpdate?

### DISCUSSION THREAD

**Discussion by ciprian-vsp** [mikro-orm/mikro-orm#4087] [Q&A] [answered]
Right now you can specify a **defaultRaw** to run your SQL (for example we want to run **NOW()** when creating an entity) and we also want to run the **NOW()** when updating the entity. 
The existing **onUpdate** method does not run the raw SQL, is there a way to specify the raw SQL for **onUpdate**? Something like **onUpdateRaw**?
Thank you!

**Comment by B4nan** [comment, marked as answer]
In MySQL you can do this:

```ts
@Property({ columnType: 'timestamp', defaultRaw: `current_timestamp` })
createdAt!: Date;

@Property({ columnType: 'timestamp', defaultRaw: `current_timestamp`, extra: `on update current_timestamp` })
updatedAt!: Date;
```

`prop.onUpdate` is what gets assigned to the entity, so it needs to be a value, not SQL fragment. Maybe we could have something like `onUpdateRaw`, mapping back the value returned by the database - although it would require additional query with MySQL as there are no returning statements. And I don't think we have implemented support for returning statements on update queries, but that sounds like a minor issue.

Anyway, what's wrong with `onUpdate: () => new Date()`?

**Reply by ciprian-vsp** [reply]
Nothing wrong with it, I wanted to be consistent with `NOW()`. I thought maybe there was a way and didn't see it in the documentation.
Forgot to mention that I am using PostgreSQL (I see that extra is only for MySQL).
I ended up using **new Date()**.
Thanks!
